tests/unit/protocol.tcl

```diff
@@ -135,6 +135,59 @@ start_server {tags {"protocol network"}} {
         assert_equal [r read] {a}
     }
 
+    test "bulk reply protocol" {
+        # value=2 (int encoding)
+        r set crlf 2
+        assert_equal [r rawread 5] "+OK\r\n"
+        r get crlf
+        assert_equal [r rawread 7] "\$1\r\n2\r\n"
+        r object encoding crlf
+        assert_equal [r rawread 9] "\$3\r\nint\r\n"
+
+        # value=9223372036854775807 (int encoding)
+        r set crlf 9223372036854775807
+        assert_equal [r rawread 5] "+OK\r\n"
+        r get crlf
+        assert_equal [r rawread 26] "\$19\r\n9223372036854775807\r\n"
+        r object encoding crlf
+        assert_equal [r rawread 9] "\$3\r\nint\r\n"
+
+        # value=-9223372036854775808 (int encoding)
+        r set crlf -9223372036854775808
+        assert_equal [r rawread 5] "+OK\r\n"
+        r get crlf
+        assert_equal [r rawread 27] "\$20\r\n-9223372036854775808\r\n"
+        r object encoding crlf
+        assert_equal [r rawread 9] "\$3\r\nint\r\n"
```

Comment:
@raffertyyu sorry for my oversight.
this test failed on 32bit because it was stored as long.
we can change them to 2147483647 and -2147483648.
```suggestion
        # value=2147483647 (int encoding)
        r set crlf 2147483647
        assert_equal [r rawread 5] "+OK\r\n"
        r get crlf
        assert_equal [r rawread 17] "\$10\r\n2147483647\r\n"
        r object encoding crlf
        assert_equal [r rawread 9] "\$3\r\nint\r\n"

        # value=-2147483648 (int encoding)
        r set crlf -2147483648
        assert_equal [r rawread 5] "+OK\r\n"
        r get crlf
        assert_equal [r rawread 18] "\$11\r\n-2147483648\r\n"
        r object encoding crlf
        assert_equal [r rawread 9] "\$3\r\nint\r\n"
```